### DISCUSSION TITLE
Is `LanguageDetector` thread-safe?

### DISCUSSION THREAD

**Discussion by Vasniktel** [pemistahl/lingua-rs#277] [Q&A] [answered]
Is it safe to call methods on the same `LanguageDetector` object from different threads?

**Comment by pemistahl** [comment, marked as answer]
Yes, it is thread-safe. Otherwise, the parallel variants of the `LanguageDetector` methods would not work because they do exactly that, namely sharing the same instance across multiple threads. 

In order to make the library thread-safe, the only thing necessary was to wrap the `HashMap`s which hold the language models in `RwLock`s.
